<?xml version="1.0"?>
<launch>
	<!-- USB port name -->
	<arg name="port" default="/adev/ttyACM0"/>
	<!-- serial baud rate -->
	<arg name="baud_rate" default="115200"/>

	<!-- base link frame -->
	<arg name="base_frame_id" default="base_link"/>

	<!-- PID constant (P) for linear X  -->
	<arg name="pid_x_p" default="120"/>
	<!-- PID constant (I) for linear X  -->
	<arg name="pid_x_i" default="1600"/>
	<!-- PID constant (D) for linear X  -->
	<arg name="pid_x_d" default="6"/>
	<!-- PID coefficient derivative filter (N) for linear X  -->
	<arg name="pid_x_n" default="50"/>
	<!-- PID high limit output for linear X  -->
	<arg name="pid_x_high" default="1000"/>
	<!-- PID low limit output for linear X  -->
	<arg name="pid_x_low" default="-1000"/>
	<!-- PID integration windup for linear X  -->
	<arg name="pid_x_windup" default="true"/>

	<!-- PID constant (P) for angular Z  -->
	<arg name="pid_z_p" default="30"/>
	<!-- PID constant (I) for angular Z  -->
	<arg name="pid_z_i" default="461.5"/>
	<!-- PID constant (D) for angular Z  -->
	<arg name="pid_z_d" default="1.3"/>
	<!-- PID coefficient derivative filter (N) for angular Z  -->
	<arg name="pid_z_n" default="50"/>
	<!-- PID high limit output for angular Z  -->
	<arg name="pid_z_high" default="1000"/>
	<!-- PID low limit output for angular Z  -->
	<arg name="pid_z_low" default="-1000"/>
	<!-- PID integration windup for angular Z  -->
	<arg name="pid_z_windup" default="true"/>

	<!-- distance between wheel (differentational robot) -->
	<arg name="wheel_distance" default="0.468"/>
	<!-- right wheel motor diameter in meter (differentational robot) -->
	<arg name="wheel_right_size" default="0.15"/>
	<!-- left wheel motor diameter in meter (differentational robot) -->
	<arg name="wheel_left_size" default="0.15"/>

	<!-- count per rotation(CPR) of right wheel encoder -->
	<arg name="encoder_wheel_right_cpr" default="3600"/>
	<!-- number of coefficient(N) derivative filter right wheel encoder -->
	<arg name="encoder_wheel_right_n" default="50"/>
	<!-- count per rotation(CPR) of left wheel encoder -->
	<arg name="encoder_wheel_left_cpr" default="3600"/>
	<!-- number of coefficient(N) derivative filter left wheel encoder -->
	<arg name="encoder_wheel_left_n" default="50"/>

	<!-- imu acceleration bias x -->
	<arg name="imu_acc_bias_x" default="0"/>
	<!-- imu acceleration bias y -->
	<arg name="imu_acc_bias_y" default="0"/>
	<!-- imu acceleration bias z -->
	<arg name="imu_acc_bias_z" default="0"/>
	<!-- imu acceleration scale x -->
	<arg name="imu_acc_scale_x" default="1"/>
	<!-- imu acceleration scale y -->
	<arg name="imu_acc_scale_y" default="1"/>
	<!-- imu acceleration scale z -->
	<arg name="imu_acc_scale_z" default="1"/>

	<!-- imu gyro bias x -->
	<arg name="imu_gyro_bias_x" default="0"/>
	<!-- imu gyro bias y -->
	<arg name="imu_gyro_bias_y" default="0"/>
	<!-- imu gyro bias z -->
	<arg name="imu_gyro_bias_z" default="0"/>

	<!-- imu mag bias x -->
	<arg name="imu_mag_bias_x" default="0"/>
	<!-- imu mag bias y -->
	<arg name="imu_mag_bias_y" default="0"/>
	<!-- imu mag bias z -->
	<arg name="imu_mag_bias_z" default="0"/>
	<!-- imu mag scale x -->
	<arg name="imu_mag_scale_x" default="1"/>
	<!-- imu mag scale y -->
	<arg name="imu_mag_scale_y" default="1"/>
	<!-- imu mag scale z -->
	<arg name="imu_mag_scale_z" default="1"/>

	<!-- advance data request for monitor only -->
	<arg name="advance" default="true"/>
	<!-- debug data receive and send for monitor only -->
	<arg name="debug" default="true"/>

	<!-- <node pkg="mrc10100" type="mrc10100_node" name="mrc10100_node" output="screen">
		<param name="port" value="$(arg port)"/>
		<param name="baud_rate" value="$(arg baud_rate)"/>
		<param name="base_frame_id" value="$(arg base_frame_id)"/>
		<param name="wheel_distance" value="$(arg wheel_distance)"/>
		<param name="wheel_right_size" value="$(arg wheel_right_size)"/>
		<param name="wheel_left_size" value="$(arg wheel_left_size)"/>
		<param name="encoder_wheel_right_cpr" value="$(arg encoder_wheel_right_cpr)"/>
		<param name="encoder_wheel_right_n" value="$(arg encoder_wheel_right_n)"/>
		<param name="encoder_wheel_left_cpr" value="$(arg encoder_wheel_left_cpr)"/>
		<param name="encoder_wheel_left_n" value="$(arg encoder_wheel_left_n)"/>
		<param name="pid_x_p" value="$(arg pid_x_p)"/>
		<param name="pid_x_i" value="$(arg pid_x_i)"/>
		<param name="pid_x_d" value="$(arg pid_x_d)"/>
		<param name="pid_x_n" value="$(arg pid_x_n)"/>
		<param name="pid_x_high" value="$(arg pid_x_high)"/>
		<param name="pid_x_low" value="$(arg pid_x_low)"/>
		<param name="pid_x_windup" value="$(arg pid_x_windup)"/>
		<param name="pid_z_p" value="$(arg pid_z_p)"/>
		<param name="pid_z_i" value="$(arg pid_z_i)"/>
		<param name="pid_z_d" value="$(arg pid_z_d)"/>
		<param name="pid_z_n" value="$(arg pid_z_n)"/>
		<param name="pid_z_high" value="$(arg pid_z_high)"/>
		<param name="pid_z_low" value="$(arg pid_z_low)"/>
		<param name="pid_z_windup" value="$(arg pid_z_windup)"/>
		<param name="imu_acc_bias_x" value="$(arg imu_acc_bias_x)"/>
		<param name="imu_acc_bias_y" value="$(arg imu_acc_bias_y)"/>
		<param name="imu_acc_bias_z" value="$(arg imu_acc_bias_z)"/>
		<param name="imu_acc_scale_x" value="$(arg imu_acc_scale_x)"/>
		<param name="imu_acc_scale_y" value="$(arg imu_acc_scale_y)"/>
		<param name="imu_acc_scale_z" value="$(arg imu_acc_scale_z)"/>
		<param name="imu_gyro_bias_x" value="$(arg imu_gyro_bias_x)"/>
		<param name="imu_gyro_bias_y" value="$(arg imu_gyro_bias_y)"/>
		<param name="imu_gyro_bias_z" value="$(arg imu_gyro_bias_z)"/>
		<param name="imu_mag_bias_x" value="$(arg imu_mag_bias_x)"/>
		<param name="imu_mag_bias_y" value="$(arg imu_mag_bias_y)"/>
		<param name="imu_mag_bias_z" value="$(arg imu_mag_bias_z)"/>
		<param name="imu_mag_scale_x" value="$(arg imu_mag_scale_x)"/>
		<param name="imu_mag_scale_y" value="$(arg imu_mag_scale_y)"/>
		<param name="imu_mag_scale_z" value="$(arg imu_mag_scale_z)"/>
		<param name="advance" value="$(arg advance)"/>
		<param name="debug" value="$(arg debug)"/>
	</node> -->
	<node pkg="mrc10100" type="mrc10100_node" name="mrc10100_node" output="screen">
		<!-- USB port name -->
		<param name="port" value="/dev/ttyACM0"/>
		<!-- serial baud rate -->
		<param name="baud_rate" value="115200"/>

		<!-- base link frame -->
		<param name="base_frame_id" value="base_link"/>

		<!-- PID constant (P) for linear X  -->
		<param name="pid_x_p" value="120"/>
		<!-- PID constant (I) for linear X  -->
		<param name="pid_x_i" value="1600"/>
		<!-- PID constant (D) for linear X  -->
		<param name="pid_x_d" value="6"/>
		<!-- PID coefficient derivative filter (N) for linear X  -->
		<param name="pid_x_n" value="50"/>
		<!-- PID high limit output for linear X  -->
		<param name="pid_x_high" value="1000"/>
		<!-- PID low limit output for linear X  -->
		<param name="pid_x_low" value="-1000"/>
		<!-- PID integration windup for linear X  -->
		<param name="pid_x_windup" value="true"/>

		<!-- PID constant (P) for angular Z  -->
		<param name="pid_z_p" value="30"/>
		<!-- PID constant (I) for angular Z  -->
		<param name="pid_z_i" value="461.5"/>
		<!-- PID constant (D) for angular Z  -->
		<param name="pid_z_d" value="1.3"/>
		<!-- PID coefficient derivative filter (N) for angular Z  -->
		<param name="pid_z_n" value="50"/>
		<!-- PID high limit output for angular Z  -->
		<param name="pid_z_high" value="1000"/>
		<!-- PID low limit output for angular Z  -->
		<param name="pid_z_low" value="-1000"/>
		<!-- PID integration windup for angular Z  -->
		<param name="pid_z_windup" value="true"/>

		<!-- distance between wheel (differentational robot) -->
		<param name="wheel_distance" value="0.468"/>
		<!-- right wheel motor diameter in meter (differentational robot) -->
		<param name="wheel_right_size" value="0.15"/>
		<!-- left wheel motor diameter in meter (differentational robot) -->
		<param name="wheel_left_size" value="0.15"/>

		<!-- count per rotation(CPR) of right wheel encoder -->
		<param name="encoder_wheel_right_cpr" value="3600"/>
		<!-- number of coefficient(N) derivative filter right wheel encoder -->
		<param name="encoder_wheel_right_n" value="50"/>
		<!-- count per rotation(CPR) of left wheel encoder -->
		<param name="encoder_wheel_left_cpr" value="3600"/>
		<!-- number of coefficient(N) derivative filter left wheel encoder -->
		<param name="encoder_wheel_left_n" value="50"/>

		<!-- imu acceleration bias x -->
		<param name="imu_acc_bias_x" value="10"/>
		<!-- imu acceleration bias y -->
		<param name="imu_acc_bias_y" value="10"/>
		<!-- imu acceleration bias z -->
		<param name="imu_acc_bias_z" value="10"/>
		<!-- imu acceleration scale x -->
		<param name="imu_acc_scale_x" value="11"/>
		<!-- imu acceleration scale y -->
		<param name="imu_acc_scale_y" value="11"/>
		<!-- imu acceleration scale z -->
		<param name="imu_acc_scale_z" value="11"/>

		<!-- imu gyro bias x -->
		<param name="imu_gyro_bias_x" value="10"/>
		<!-- imu gyro bias y -->
		<param name="imu_gyro_bias_y" value="10"/>
		<!-- imu gyro bias z -->
		<param name="imu_gyro_bias_z" value="10"/>

		<!-- imu mag bias x -->
		<param name="imu_mag_bias_x" value="10"/>
		<!-- imu mag bias y -->
		<param name="imu_mag_bias_y" value="10"/>
		<!-- imu mag bias z -->
		<param name="imu_mag_bias_z" value="10"/>
		<!-- imu mag scale x -->
		<param name="imu_mag_scale_x" value="11"/>
		<!-- imu mag scale y -->
		<param name="imu_mag_scale_y" value="11"/>
		<!-- imu mag scale z -->
		<param name="imu_mag_scale_z" value="11"/>

		<!-- advance data request for monitor only -->
		<param name="advance" value="true"/>
		<!-- debug data receive and send for monitor only -->
		<param name="debug" value="true"/>
	</node>

</launch>